### ISSUE TITLE
Problem with policies when your server had shut down incorrectly

### ISSUE THREAD

**Issue by ivanova-zz** [ClickHouse/ClickHouse#24260]
The server had shut down incorrectly due to a power problem

Our storage_configuration:
```
<storage_configuration>
    <disks>
        <disk_ssd>
	    <path>/data/fast/</path>
	</disk_ssd>
        <disk_hdd> 
            <path>/data/slow/</path>
	    <keep_free_space_bytes>10485760</keep_free_space_bytes>
        </disk_hdd> 
    </disks>

    <policies>
        <moving_from_ssd_to_hdd>
            <volumes>
                <hot>
                    <disk>disk_ssd</disk>
                    <max_data_part_size_bytes>1073741824</max_data_part_size_bytes>
                </hot>
                <cold>
                    <disk>disk_hdd</disk>
                </cold>
            </volumes>
            <move_factor>0.3</move_factor>
        </moving_from_ssd_to_hdd>
    </policies>
</storage_configuration>
```

For data recovery I used the flag /var/lib/clickhouse/flags/force_restore_data . I created it by the command: 
```$ sudo -u clickhouse touch /var/lib/clickhouse/flags/force_restore_data```
 

When I launched clickhouse-server after the server came back to work I got the error:
> 2021.05.17 11:18:02.532039 [ 12184 ] {} <Error> Application: DB::Exception: Part 20210513_3835188_3934695_149 already exists: Cannot attach table `stats635`.`events12001` from metadata file /var/lib/clickhouse/store/682/6821e6fa-4f0e-4a4b-80b2-d3b2efb0dcfe/events12001.sql from query ATTACH TABLE stats635.events12001 UUID 'fe3ca1ac-caa9-4b2f-93be-3b2b965047db' (`date` Date DEFAULT toDate(time), `time` DateTime, `msec` UInt64, `value` Float32 DEFAULT CAST(1, 'Float32'), `ip` String, `country` String, `city` String, `region` String, `isp` String, `trackHost` String, `optimalConnectID` String, `isOptimal` String, `eventid` String, `serverType` String, `appVersion` String, `serverLocation` String, `appType` String, `installDate` String, `isHiddenApp` String, `externalIP` String, `connectEventID` String, `timeConnecting` String, `numberLaunch` String, `uuid` String, `appName` String, `networkType` String, `idfa` String, `IPAddress` String, `type` String, `countryCode` String, `asn` String, `aso` String, `hoster` String, `requestType` String, `reTryRequest` String, `connectionType` String, `UUIDOrigin` String) ENGINE = MergeTree PARTITION BY toYYYYMMDD(time) ORDER BY date TTL date TO VOLUME 'hot', date + toIntervalDay(7) TO VOLUME 'cold' SETTINGS index_granularity = 8192, storage_policy = 'moving_from_ssd_to_hdd': while loading database `stats635` from path /var/lib/clickhouse/metadata/stats635
> 2021.05.17 11:18:02.532087 [ 12184 ] {} <Information> Application: shutting down
> 2021.05.17 11:18:02.532094 [ 12184 ] {} <Debug> Application: Uninitializing subsystem: Logging Subsystem
> 2021.05.17 11:18:02.532196 [ 12185 ] {} <Trace> BaseDaemon: Received signal -2
> 2021.05.17 11:18:02.532246 [ 12185 ] {} <Information> BaseDaemon: Stop SignalListener thread
> 2021.05.17 11:18:02.577780 [ 12183 ] {} <Information> Application: Child process exited normally with code 70.

Then I checked for the presence of a folder named "20210513_3835188_3934695_149" under the path /data/fast/XXX/ and under the path /data/slow/XXX/. 
The folder was in both paths, which was the cause of this error. I checked for content differences between these folders - there were no differences. 
Then I moved the folder from /data/fast/XXX/ and started clickhouse-server. 
And It started working fine!!! :)

It seems to me that the cause of this problem was a power cut at the time of transferring data from ssd to hdd. Are there any other solutions to this problem?

**Comment by alesapin** [comment]
Yes it looks like bug and we even have a comment in the code:  https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/MergeTree/MergeTreePartsMover.cpp#L255.  So seems like your server was powered off exactly on this line. Maybe we can do atomic rename of two directories...

**Comment by alexandrpaliy** [comment]
Faced the same issue some time ago, wrote a simple bash script to remove all duplicates from slow data dir:

```
#!/usr/bin/bash
DIR_DATA_FAST="/path/to/data/fast/dir"
DIR_DATA_SLOW="/path/to/data/slow/dir"

for partname in $(find "${DIR_DATA_FAST}/store" -maxdepth 3 -mindepth 3 -type d -regextype sed -regex ".*/.*/[0-9]\{8\}_.*" -printf "%P\n")
do
  if [ -d "${DIR_DATA_SLOW}/store/${partname}" ]
  then
    echo "DIR ${DIR_DATA_SLOW}/store/${partname} exists !!! removing"
    rm -rf "${DIR_DATA_SLOW}/store/${partname}"
  fi
done
```

Message in russian ClickHouse telegram channel for reference: https://t.me/clickhouse_ru/246484
This script solved the problem in my case, but **use it on your own risk**. You can comment "rm" line, so script will just print out duplicates' names and not modify anything at all.
